test/forms/ExciseUnderpaymentFormProviderSpec.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import base.SpecBase
+import mocks.config.MockAppConfig
+import models.UnderpaymentAmount
+import play.api.data.{Form, FormError}
+
+class ExciseUnderpaymentFormProviderSpec extends SpecBase {
+
+  private final val originalKey = "original"
+  private final val amendedKey = "amended"
+  private final val originalNonNumberMessageKey = "ExciseUnderpayment.error.originalNonNumber"
+  private final val amendedNonNumberMessageKey = "ExciseUnderpayment.error.amendedNonNumber"
+  private final val originalNonEmptyMessageKey = "ExciseUnderpayment.error.originalNonEmpty"
+  private final val amendedNonEmptyMessageKey = "ExciseUnderpayment.error.amendedNonEmpty"
+  private final val fifty = "50"
+  private final val forty = "40"
+  private final val nonNumeric = "@£$%FGB"
+
+  def formBuilder(original: String = "", amended: String = ""): Map[String, String] = Map(
+    originalKey -> original,
+    amendedKey -> amended
+  )
+
+  def formBinder(formValues: Map[String, String] = Map(originalKey -> "", amendedKey -> "")): Form[UnderpaymentAmount] =
+    new ExciseUnderpaymentFormProvider()(MockAppConfig).apply().bind(formValues)
+
+  "Binding a form with invalid data" when {
+
+    "no values provided" should {
+      "result in a form with errors" in {
+        formBinder().errors mustBe Seq(
+          FormError(originalKey, originalNonEmptyMessageKey),
+          FormError(amendedKey, amendedNonEmptyMessageKey)
+        )
+      }
+
```

Comment:
I think you may have got a missing brace here which has accidentally been put at line 98